src/Encamina.Enmarcha.SemanticKernel/MemoryStoreExtender.cs

```diff
@@ -0,0 +1,165 @@
+    /// <param name="logger">Log service.</param>
+    public MemoryStoreExtender(IMemoryStore memoryStore, ILogger<MemoryStoreExtender> logger)
+    {
+        MemoryStore = memoryStore;
+        this.logger = logger;
+    }
+
+    /// <inheritdoc/>
+    public event EventHandler<MemoryStoreEventArgs> MemoryStoreEvent;
+
+    /// <inheritdoc/>
+    public IMemoryStore MemoryStore { get; init; }
+
+    /// <inheritdoc/>
+    public void RaiseMemoryStoreEvent(MemoryStoreEventArgs e) => MemoryStoreEvent?.Invoke(this, e);
+
+    /// <inheritdoc/>
+    public virtual async Task UpsertMemoryAsync(string memoryId, string collectionName, IEnumerable<string> chunks, Kernel kernel, IDictionary<string, string> metadata = null, CancellationToken cancellationToken = default)
+    {
+        await DeleteMemoryAsync(memoryId, collectionName, cancellationToken);
+        await SaveChunks(memoryId, collectionName, chunks, metadata, kernel, cancellationToken);
+    }
+
+    /// <inheritdoc/>
+    public virtual async Task DeleteMemoryAsync(string memoryId, string collectionName, CancellationToken cancellationToken)
+    {
+        var chunkSize = await GetChunkSize(memoryId, collectionName, cancellationToken);
+        if (chunkSize > 0)
+        {
+            var keys = Enumerable.Range(0, chunkSize).Select(i => BuildMemoryIdentifier(memoryId, i));
+            await MemoryStore.RemoveBatchAsync(collectionName, keys, cancellationToken);
+            RaiseMemoryStoreEvent(new() { EventType = MemoryStoreEventTypes.RemoveBatch, Keys = keys, CollectionName = collectionName });
+        }
+    }
+
+    /// <inheritdoc/>
+    public virtual async Task<MemoryContent> GetMemoryAsync(string memoryId, string collectionName, CancellationToken cancellationToken)
+    {
+        var chunkSize = await GetChunkSize(memoryId, collectionName, cancellationToken);
+
+        if (chunkSize == 0)
+        {
+            return null;
+        }
+
+        var keys = Enumerable.Range(0, chunkSize).Select(i => BuildMemoryIdentifier(memoryId, i));
+        var memoryRecords = await MemoryStore.GetBatchAsync(collectionName, keys, cancellationToken: cancellationToken).ToListAsync(cancellationToken);
+        RaiseMemoryStoreEvent(new() { EventType = MemoryStoreEventTypes.GetBatch, Keys = keys, CollectionName = collectionName });
+
+        return new MemoryContent
+        {
+            Metadata = JsonSerializer.Deserialize<Dictionary<string, string>>(memoryRecords[0].Metadata.AdditionalMetadata),
+            Chunks = memoryRecords.Select(m => m.Metadata.Text),
+        };
+    }
+
+    /// <inheritdoc/>
+    public virtual async IAsyncEnumerable<string> BatchUpsertMemoriesAsync(string collectionName, IDictionary<string, MemoryContent> memoryContents, Kernel kernel, [EnumeratorCancellation] CancellationToken cancellationToken)
+    {
+        var memoryRecords = new Collection<MemoryRecord>();
+        var memoryIds = new List<string>();
```

Comment:
This variable is unused. You should remove it.

Reply:
That's right, I got it from a test. I'm deleting it